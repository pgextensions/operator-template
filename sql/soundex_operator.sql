CREATE EXTENSION "soundex-operator";

-- john and joan should be equal
SELECT soundex_eq('john', 'joan');

-- bob and shirley should not
SELECT soundex_eq('john', 'shirley');

-- and the inverse of each
SELECT soundex_ne('john', 'joan');
SELECT soundex_ne('john', 'shirley');

-- those should translate into the operators
SELECT 'john' >@@< 'joan';
SELECT 'john' >@@< 'shirley';

SELECT 'john' <@@> 'joan';
SELECT 'john' <@@> 'shirley';
