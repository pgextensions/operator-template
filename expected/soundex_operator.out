CREATE EXTENSION "soundex-operator";
-- john and joan should be equal
SELECT soundex_eq('john', 'joan');
 soundex_eq 
------------
 t
(1 row)

-- bob and shirley should not
SELECT soundex_eq('john', 'shirley');
 soundex_eq 
------------
 f
(1 row)

-- and the inverse of each
SELECT soundex_ne('john', 'joan');
 soundex_ne 
------------
 f
(1 row)

SELECT soundex_ne('john', 'shirley');
 soundex_ne 
------------
 t
(1 row)

-- those should translate into the operators
SELECT 'john' >@@< 'joan';
 ?column? 
----------
 t
(1 row)

SELECT 'john' >@@< 'shirley';
 ?column? 
----------
 f
(1 row)

SELECT 'john' <@@> 'joan';
 ?column? 
----------
 f
(1 row)

SELECT 'john' <@@> 'shirley';
 ?column? 
----------
 t
(1 row)

